src/reporter/index.js

```diff
@@ -168,90 +188,162 @@ export default class Reporter {
         });
     }
 
-    _assignTaskEventHandlers () {
-        const task = this.task;
+    async dispatchToPlugin ({ method, args = [] }) {
+        try {
+            await this.plugin[method](...args);
+        }
+        catch (originalError) {
+            const uncaughError = new ReporterPluginError({
+                name: typeof this.plugin.name === 'string' ? this.plugin.name : 'customReporter',
```

Comment:
We discussed that `Function.name` could be used as a plugin name when a reporter is constructed from a factory function. If the factory function doesn't have a name, we need to fill a placeholder (smth like `unnamed function` or `function () {}`).